CONTRIBUTING.md

```diff
@@ -0,0 +1,18 @@
+# Contributing to this project/repository
+To contribute is fairly easy:
+
+Until a proper RISC OS git tool is being released, we encourage you to use for your "local git" either a Windows computer or a Linux computer or Virtual Machine.
+
+There are some very useful tips [here at the ROOL Forum](https://www.riscosopen.org/content/documents/git-cheatsheet)
+
+## General guidelines
+In general you'd want to:
+- Git clone a repo from here to your local git
+- Create your branch using git branch (usually for branching naming you'd want to create the branch name as: <your nick>-<what type of changes you are applying>)
```

Comment:
Is this right? Wouldn't you fork and the clone from there, then push to your fork and do a pull request? That's how we do it on the ROOL gitlab anyway.

Reply:
We can either fork and clone, but if we use branches is quicker to submit a pull request , just by experience. If we prefer forks then that's good for me. 

For the branching, just:

- git clone <repo>
- cd <repo>
- git branch mybranch-doing-this-change
- code / change things / run tests / git add+commit / git push -u origin mybranch-doing-this-change
- When you're finally done, just open a pull request either via git (CLI) or here on github (WEB) and tag the code-reviews-team 
- wait for the storm of comments, fix your code in your branch as required
- git add+commit / git push -u origin mybranch-doing-this-change
- resubmit your pull request (as I did for the code of conduct changes) so we have full history of those changes in a single place and with fixes
- done

But I am open to suggestions if we want to use a different approach :)

Reply:
For the "Do you need to do something with keys?"

I strongly recommend to use RSA keys (just add your public key if you haven't done it yet) that makes pushes quicker and also safer. Passwords are historically weak and, even if this is an Open Source community, you do want to avoid intruders to be able to impersonate you. 

If we all agree on using RSA keys I will add the procedure OR the links to the github procedures in the CONTRINUTING.md.

I am sorry I code for big companies so git is something I use on a daily base and like constantly, so I did not think of adding more details, my bad. Thanks for pointing it out btw!

Reply:
Ok added more details and background info so we can decide what is best.
Guys please let me know if we prefer Fork and branch in the fork or if we prefer to branch from the codebase. I am happy with both, thanks!
@skymandr @InteractiveYelp ^^^